config/src/main/java/org/springframework/security/config/annotation/web/configurers/JeeConfigurer.java

```diff
@@ -100,9 +101,7 @@ public JeeConfigurer() {
 	 */
 	public JeeConfigurer<H> mappableAuthorities(String... mappableRoles) {
 		this.mappableRoles.clear();
-		for (String role : mappableRoles) {
-			this.mappableRoles.add(role);
-		}
+		this.mappableRoles.addAll(Arrays.asList(mappableRoles));
```

Comment:
This allocates an additional object which is not preferred since it in turn will need garbage collected.

Reply:
You're right. Unfortunately `HashSet` just uses looped `add` calls in it's `addAll` implementation.

Generally `addAll` has the benefit that the number of elements which will be added is known in advance, so the internal data structure can be resized to the correct number. Array-based collections like `ArrayList` can use `System.arraycopy` internally, while looped `add` calls may require multiple internal resizes.